crates/hir/src/lib.rs

```diff
@@ -1613,7 +1614,7 @@ impl Impl {
 
     // FIXME: the return type is wrong. This should be a hir version of
     // `TraitRef` (ie, resolved `TypeRef`).
-    pub fn target_trait(self, db: &dyn HirDatabase) -> Option<TypeRef> {
+    pub fn target_trait(self, db: &dyn HirDatabase) -> Option<TraitRef> {
         db.impl_data(self.id).target_trait.clone()
     }
```

Comment:
Just in general for a future PR, this should return some kind of trait thing that knows about the self type right? or should this just return `hir::Trait`?

Reply:
Probably a HIR wrapper of `TraitRef`, i.e. what `Type` is to `Ty`, would make sense. As far as I can see this is only used in one place currently, which just checks whether it's `Some`, so we're free to make it whatever makes most sense as an API. We could also have `target_trait` and `target_trait_ref`.

Reply:
We also should align this with ast naming, as it feels better: https://github.com/matklad/rust-analyzer/blob/b352b90c86e96fe1416f4527c1c88dc2c2c17451/crates/syntax/src/ast/node_ext.rs#L248-L261

Reply:
Actually, we should aligh this across the code base, in item tree and in hir_def as well.

Reply:
>Probably a HIR wrapper of TraitRef, i.e. what Type is to Ty, would make sense. 

I've confused myself over this, so let me clarify: a wrapper around `hir_ty::TraitRef`, not around `hir_def::Trait`.